docs/src/language_servers.md

```diff
@@ -778,35 +778,90 @@ See [Javascript/TypeScript](#javascripttypescript).
-2. Download the [typst-lsp](https://github.com/nvarner/typst-lsp/releases) language server executable for your platform.
-3. Open `Preferences > Package Settings > LSP > Settings` and add the `"typst-lsp"` client configuration to the `"clients"`:
+2. Optional: to enable auto-completions for the relevant situations in Typst files, adjust Sublime's `"auto_complete_selector"` and/or `"auto_complete_triggers"` setting (`Preferences > Settings`); for example
+
+    ```jsonc
+    {
+        "auto_complete_triggers":
+        [
+            {"selector": "text.html, text.xml", "characters": "<"},
+            {"selector": "punctuation.accessor", "rhs_empty": true},
+            {"selector": "text.typst", "characters": "#", "rhs_empty": true},
+        ],
+    }
+    ```
+
+There are 2 available languages servers.
+
+### Tinymist
+
+This server has more features, like go to definition, rename, etc.
+
+1. Install [tinymist](https://github.com/Myriad-Dreamin/tinymist).
+2. Open `Preferences > Package Settings > LSP > Settings` and add the `"tinymist"` client configuration to the `"clients"`:
 
     ```jsonc
     {
         "clients": {
-            "typst-lsp": {
+            "tinymist": {
                 "enabled": true,
-                "command": ["C:\\path\\to\\typst-lsp-win32-x64.exe"],  // adjust this path according to your platform/setup
-                "selector": "text.typst"
+                "command": ["path/to/tinymist"],  // adjust this path according to your platform/setup
+                "selector": "text.typst",
+                // you can provide some initialization options:
+                "initializationOptions": {
+                    "exportPdf": "never",
+                    "typstExtraArgs": [],
+                },
             }
         }
     }
     ```
 
-4. Optional: to enable auto-completions for the relevant situations in Typst files, adjust Sublime's `"auto_complete_selector"` and/or `"auto_complete_triggers"` setting (`Preferences > Settings`); for example
+3. Optional: to enable some useful commands provided by language server, add the following to the `*.sublime-commands`:
+
+    <!-- how to call: see https://github.com/Myriad-Dreamin/tinymist/blob/main/editors/vscode/src/extension.ts -->
+    ```jsonc title="Packages/User/Default.sublime-commands"
+    [
+        // ...
+        {
+            "caption": "tinymist - Pin the main file to the currently opened document",
+            "command": "lsp_execute",
+            "args": {
+                "session_name": "tinymist",
+                "command_name": "tinymist.doPinMain",
+                "command_args": ["${file}"]
+            }
+        },
+        {
+            "caption": "tinymist - Unpin the main file",
+            "command": "lsp_execute",
+            "args": {
+                "session_name": "tinymist",
+                "command_name": "tinymist.doPinMain",
+                "command_args": [null]
+            }
+        },
+    ]
+    ```
```

Comment:
This seems like a copy-paste from the `Typst-lsp` section with just the server name changed. Are you sure that `Tinymist` supports same commands?

Reply:
Yes. Also, the argument for the second command is different

Reply:
Then why can I not find any references to `doPinMain` in its repository (https://github.com/Myriad-Dreamin/tinymist/blob/main/editors/vscode/src/extension.ts)

Reply:
Thanks for noticing, I fixed this in my config, but forgot about documentation

Reply:
https://github.com/Myriad-Dreamin/tinymist/blob/ee50b24ce43d98d560e38c49ad8c34769a97a64e/crates/tinymist/src/server.rs#L262

@istudyatuni it is a lsp command provided by server.

```rs
.with_command("tinymist.pinMain", State::pin_document)
```